generators/server/templates/src/main/java/package/security/oauth2/AuthorizationHeaderUtil.java.ejs

```diff
@@ -18,46 +18,158 @@
+                return Optional.of(authorizationHeaderValue);
+            }
 
-        if (accessToken == null) {
-            return Optional.empty();
-        } else {
-            String tokenType = "Bearer";
-            String authorizationHeaderValue = String.format("%s %s", tokenType, accessToken);
+        } else if (authentication instanceof JwtAuthenticationToken) {
+            JwtAuthenticationToken accessToken = (JwtAuthenticationToken) authentication;
+            String tokenValue = accessToken.getToken().getTokenValue();
+            String authorizationHeaderValue = String.format("%s %s", OAuth2AccessToken.TokenType.BEARER.getValue(), tokenValue);
             return Optional.of(authorizationHeaderValue);
         }
+        return Optional.empty();
+    }
+
+    private String refreshToken(OAuth2AuthorizedClient client, OAuth2AuthenticationToken oauthToken) {
+        OAuth2AccessTokenResponse atr = refreshTokenClient(client);
+        if (atr == null || atr.getAccessToken() == null) {
+            log.info("Failed to refresh token for user");
+            return null;
+        }
+
+        OAuth2RefreshToken refreshToken = atr.getRefreshToken() != null? atr.getRefreshToken(): client.getRefreshToken();
+        OAuth2AuthorizedClient updatedClient = new OAuth2AuthorizedClient(
+            client.getClientRegistration(),
+            client.getPrincipalName(),
+            atr.getAccessToken(),
+            refreshToken
+        );
+
+        clientService.saveAuthorizedClient(updatedClient, oauthToken);
+        return atr.getAccessToken().getTokenValue();
+    }
+
+    private OAuth2AccessTokenResponse refreshTokenClient(OAuth2AuthorizedClient currentClient ) {
+
+        MultiValueMap<String, String> formParameters = new LinkedMultiValueMap<>();
+        formParameters.add(OAuth2ParameterNames.GRANT_TYPE, AuthorizationGrantType.REFRESH_TOKEN.getValue());
+        formParameters.add(OAuth2ParameterNames.REFRESH_TOKEN, currentClient.getRefreshToken().getTokenValue());
+        formParameters.add(OAuth2ParameterNames.CLIENT_ID, currentClient.getClientRegistration().getClientId());
+        RequestEntity requestEntity = RequestEntity
+            .post(URI.create(currentClient.getClientRegistration().getProviderDetails().getTokenUri()))
+            .contentType(MediaType.APPLICATION_FORM_URLENCODED)
+            .body(formParameters);
+        try {
+            RestTemplate r = restTemplate(currentClient.getClientRegistration().getClientId(), currentClient.getClientRegistration().getClientSecret());
+            ResponseEntity<OAuthIdpTokenResponseDTO> responseEntity = r.exchange(requestEntity, OAuthIdpTokenResponseDTO.class);
+            return toOAuth2AccessTokenResponse(responseEntity.getBody());
+        } catch (OAuth2AuthorizationException e) {
+            log.error("Unable to refresh token ${e.getError().getErrorCode()}");
```

Comment:
do you mean that it would be correctly evaluated at run time `${e.getError().getErrorCode()}` ?

Reply:
Fixed

Reply:
@Tcharl, did you push changes?

Reply:
Oups :-).